File: in_Java/0896 Monotonic Array.java
class Solution {
    public boolean isMonotonic(int[] A) {
        int p = 1;
        while(p < A.length && A[p] == A[0]) p++;
        if(p == A.length) return true;
        boolean increase = A[p] > A[0];
        boolean decrease = A[p] < A[0];
        for(int i = 1; i < A.length; i++){
            if(increase && A[i] < A[i-1]) return false;
            if(decrease && A[i] > A[i-1]) return false;
        }
        return true;
    }
}

File: in_Python/0896 Monotonic Array.py
class Solution(object):
    def isMonotonic(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        diff = []
        for i in range(1, len(A)):
            if A[i] > A[i-1]:
                diff.append(1)
            elif A[i] == A[i-1]:
                diff.append(0)
            else:
                diff.append(-1)
        if 1 in diff and -1 in diff:
            return False
        return True